src/main/java/com/wanderersoftherift/wotr/datagen/ModRiftThemeTagsProvider.java

```diff
@@ -0,0 +1,26 @@
+package com.wanderersoftherift.wotr.datagen;
+
+import com.wanderersoftherift.wotr.WanderersOfTheRift;
+import com.wanderersoftherift.wotr.init.ModRiftThemes;
+import com.wanderersoftherift.wotr.init.ModTags;
+import com.wanderersoftherift.wotr.world.level.levelgen.theme.RiftTheme;
+import net.minecraft.core.HolderLookup;
+import net.minecraft.data.PackOutput;
+import net.minecraft.data.tags.TagsProvider;
+import net.minecraft.tags.TagEntry;
+
+import java.util.concurrent.CompletableFuture;
+
+public class ModRiftThemeTagsProvider extends TagsProvider<RiftTheme> {
+    // Get parameters from the `GatherDataEvent`s.
+    public ModRiftThemeTagsProvider(PackOutput output, CompletableFuture<HolderLookup.Provider> lookupProvider) {
+        // Second parameter is the registry key we are generating the tags for.
+        super(output, ModRiftThemes.RIFT_THEME_KEY, lookupProvider, WanderersOfTheRift.MODID);
+    }
+
+    @Override
+    protected void addTags(HolderLookup.Provider lookupProvider) {
+        tag(ModTags.RiftThemes.RANDOM_SELECTABLE).add(TagEntry.optionalElement(WanderersOfTheRift.id("cave")))
+                .add(TagEntry.optionalElement(WanderersOfTheRift.id("forest")));
```

Comment:
Should probably contain the mushroom too.

Reply:
What is the condition for a theme being included in the creative menu generated key? Maybe I should remove forest?

Reply:
I mean, maybe the creative menu should always be processor theme? Might help the builders.

Other than that, no real condition, I would primarily split processor from the list. We have a nether theme now too, unsure if that should go in or not.